packages/liferay-npm-scripts/src/utils/generateSoyDependencies.js

```diff
@@ -34,7 +37,7 @@ function generateSoyDependencies(dependencies) {
 			return resolvedDependency;
 		})
 		.filter(Boolean)
-		.filter(dependencyPath => dependencyPath !== cwd);
+		.filter(dependencyPath => !dependencyPath.endsWith(projectName));
```

Comment:
My only concern here is that there might be some project where the dependency does end with `projectName` but it doesn't correspond exactly to the basename (eg. `projectName` might be `bar`, and `dependencyPath` might be `foo-bar`). So to be fully robust the test should be:

```js
filter(dependencyPath => path.basename(dependencyPath) !== projectName);
```
```

Reply:
Good catch